crypto/rsa/rsa_gen.c

```diff
@@ -253,15 +254,26 @@ static int rsa_builtin_keygen(RSA *rsa, int bits, int primes, BIGNUM *e_value,
              * 1. 1024-bit factors are effcient when using 3072 and 4096-bit key
              * 2. stay the same logic with normal 2-prime key
              */
+            bitse -= bitsr[i];
+            if (!BN_GENCB_call(cb, 2, n++))
+                goto err;
             if (primes > 4) {
                 if (bitst < 0x9)
                     adj++;
                 else
                     adj--;
```

Comment:
please add an assertion here, that `adj >= 0 && adj <= 1`

Reply:
Be careful about adding assertions to the library.  They are something we try to avoid now.

Reply:
I agree to not add assertions here...